pyccel/parser/semantic.py

```diff
@@ -2925,7 +2924,12 @@ def _visit_FunctionCall(self, expr):
         except RuntimeError:
             pass
 
-        func     = self.scope.find(name, 'functions')
+        func = self.scope.find(name, 'functions')
+
+        if func is None:
+            name = str(expr.funcdef)
+            if name in builtin_functions_dict:
+                func = PyccelFunctionDef(name, builtin_functions_dict[name])
```

Comment:
Could you explain this change to me? How is this related to the next modification in the same function?

Reply:
When we translate a `FunctionCall` such as `math.cos(4.0)` we first search for a function definition in the scope. We find a `PyccelFunctionDef` and handle this call in `_handle_function`. `_handle_function` does the same thing as `builtin_function`. It splits the `FunctionCallArguments` in to `args` and `kwargs` and calls the class constructor. Unlike `builtin_function` it additionally wraps the constructor call in a try/except so that context clues can be added to the error message. By packing the builtin function found in `builtin_functions_dict` into a `PyccelFunctionDef` we can use the same code to handle built-in Python functions and imported functions